src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts

```diff
@@ -58,6 +57,33 @@ export class EditCellAction extends ToggleableAction {
 	}
 }
 
+export class SplitCellAction extends CellActionBase {
+	public cellType: CellType;
+
+	constructor(
+		id: string,
+		label: string,
+		cssClass: string,
+		@INotificationService notificationService: INotificationService,
+		@INotebookService private notebookService: INotebookService,
+	) {
+		super(id, label, cssClass, notificationService);
+		this._cssClass = cssClass;
+		this._tooltip = label;
+		this._label = '';
+	}
+	doRun(context: CellContext): Promise<void> {
+		let model = context.model;
+		let index = model.cells.findIndex((cell) => cell.id === context.cell.id);
+		if (context instanceof CellContext) {
```

Comment:
Why is this check necessary? You already define the parameter as CellContext and I'm curious why it would be anything else.

Reply:
@halerankin - can you comment on this?

Reply:
Yes, go ahead and remove that check.

Reply:
Done